Sming/third-party/.patches/umm_malloc.patch

```diff
@@ -16,12 +16,12 @@ index fcc4fb1..661b94a 100644
 +
 +#ifdef __ets__
 +// ESP specific code
++#include "debug_progmem.h"
 +extern void ets_intr_lock();
 +extern void ets_intr_unlock();
-+extern void ets_printf(const char*, ...);
 +
-+#define printf ets_printf
-+#define puts(...) ets_printf("%s",...)
++#define printf debug_i
```

Comment:
The debug_i macro uses printf_P_stack which does not involve any heap alloc... But I also see your concern. Will remove this and make a separate PR Then we can have better confidence. Will then test on my application which does SSL / websockets etc and measure some statistics.